### PR TITLE
Implement basic login

### PR DESCRIPTION
**Pull request by tunglt-0058**

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = "Invalid email/password combination"
+      render "new"
+    end
+  end
+
+  def destroy
+    log_out
```
Comment (DuongNT1512) [inline app/controllers/sessions_controller.rb]: check login trước khi logout nhé

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
```
Comment (DuongNT1512) [inline app/helpers/sessions_helper.rb]: current_user.present?

File: app/models/user.rb [hunk 3]
```diff
@@ -10,6 +10,11 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  def User.digest string
```
Comment (DuongNT1512) [inline app/models/user.rb]: class method nhé

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -5,7 +5,24 @@
...
+          <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+              Account <b class="caret"></b>
+            </a>
+            <ul class="dropdown-menu">
+              <li><%= link_to "Profile", current_user %></li>
+              <li><%= link_to "Settings", "#" %></li>
+              <li class="divider"></li>
+              <li>
+                <%= link_to "Log out", logout_path, method: "delete" %>
```
Comment (DuongNT1512) [inline app/views/layouts/_header.html.erb]: method: :delete

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = "Invalid email/password combination"
+      render "new"
```
Comment (DuongNT1512) [inline app/controllers/sessions_controller.rb]: render :new

File: app/models/user.rb [hunk 6]
```diff
@@ -10,6 +10,14 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  class << self
+    def digest string
+        cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
```
Comment (DuongNT1512) [inline app/models/user.rb]: indent

File: test/test_helper.rb [hunk 7]
```diff
@@ -4,5 +4,8 @@
 
 class ActiveSupport::TestCase
   fixtures :all
-  include ApplicationHelper
+
+  def is_logged_in?
+    !session[:user_id].nil?
```
Comment (DuongNT1512) [inline test/test_helper.rb]: session[:user_id].present?

**Comment by DuongNT1512** [comment]
OK

File: config/routes.rb [hunk 8]
```diff
@@ -1,10 +1,15 @@
...
   get "/about", to: "static_pages#about"
   get "/contact", to: "static_pages#contact"
   get "/signup", to: "users#new"
   post "/signup", to: "users#create"
+  get "/login", to: "sessions#new"
+  post "/login", to: "sessions#create"
+  delete "/logout", to: "sessions#destroy"
 
-  resources :users, only: :show
+  resources :users
```
Comment (hellovietnam93) [inline config/routes.rb]: có destroy chưa em?

**Pull request merged**